internal/services/containers/kubernetes_addons.go

```diff
@@ -495,11 +503,13 @@ func flattenKubernetesAddOns(profile map[string]managedclusters.ManagedClusterAd
 			}
 		}
 
+		useAADAuth := kubernetesAddonProfilelocateInConfig(omsAgent.Config, "useAADAuth")
```

Comment:
We should always set a value into state and since this would default to false it should be
```suggestion
                useAADAuth := false
		if v := kubernetesAddonProfilelocateInConfig(omsAgent.Config, "useAADAuth"); v != "" {
			useAADAuth = v
		}
```